cmd/optimizer-nri-plugin/main.go

```diff
@@ -178,11 +211,74 @@ func (p *plugin) StartContainer(_ *api.PodSandbox, container *api.Container) err
 		return err
 	}
 
+	imageRepo := container.Annotations[imageNameLabel]
+	serverURL := fmt.Sprintf("%s:%s", cfg.Domain, cfg.Port)
+
+	go func() {
+		time.Sleep(1 * time.Minute)
+		if err := sendToServer(imageRepo, persistFile, containerName, serverURL); err != nil {
+			log.WithError(err).Error("failed to send prefetch to http server")
+		}
+	}()
+
 	globalFanotifyServer[imageName] = fanotifyServer
 
 	return nil
 }
 
+func sendToServer(imageName, prefetchlistPath, containerName, serverURL string) error {
+	data, err := os.ReadFile(prefetchlistPath)
+	if err != nil {
+		return fmt.Errorf("error reading file: %w", err)
+	}
+
+	filePaths := strings.Split(string(data), "\n")
+
+	var prefetchFiles []PrefetchFile
+	for _, path := range filePaths {
+		if path != "" {
+			prefetchFiles = append(prefetchFiles, PrefetchFile{Path: path})
+		}
+	}
+
+	item := CacheItem{
+		ImageName:     imageName,
+		ContainerName: containerName,
+		PrefetchFiles: prefetchFiles,
+	}
+
+	url := fmt.Sprintf("http://%s%s", serverURL, defaultEndpoint)
+
+	err = postRequest(item, url)
+	if err != nil {
+		return fmt.Errorf("error uploading to server: %w", err)
+	}
+
+	return nil
+}
+
+func postRequest(item CacheItem, endpoint string) error {
+	data, err := json.Marshal(item)
+	if err != nil {
+		return err
+	}
+
+	resp, err := http.Post(endpoint, "application/json", bytes.NewBuffer(data))
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	fmt.Println("Server Response:", string(body))
```

Comment:
Should check the HTTP status code and return an error if the status code is invalid.